src/transformStateWithClones.js

```diff
@@ -6,8 +6,45 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    const { type } = action;
+
+    switch (type) {
+      case 'addProperties':
+        const { extraData } = action;
+
+        currentState = {
+          ...currentState, ...extraData,
+        };
+        break;
+
+      case 'removeProperties':
+        const { keysToRemove } = action;
+        const newState = { ...currentState };
+
+        for (const key of keysToRemove) {
+          delete newState[key];
+        }
+        currentState = newState;
+        break;
+
+      case 'clear':
+        currentState = {};
+        break;
+
+      default:
+        break;
+    }
+
+    result.push({ ...currentState });
```

Comment:
does tests allow you to `push(currentState)` ? just without redundant destructuring

Reply:
@Andriy-Shymkiv nope, without push(currentState) it's 11 failed =)